### ISSUE TITLE
Give the home page some design love

### ISSUE THREAD

**Issue by GarthDB** [mustache/mustache.github.com#57]
Hey @locks could you post some screen shots of websites you like and we can start some moodboarding on github issues.

**Comment by locks** [comment]
To complement the e-mail above, here's what I'd ideally like to see:
- Homepage
- Status board
- Documentation
- Implementation-specific page. These would be, for example, themed with the implementation language associated colors (Ruby -> red, JavaScript -> yellow(?)).
- Logo — I'm waiting to hear back from an artist.

How the first three points are structured is up for debate.

**Comment by una** [comment]
Current status as a reference:

![screen shot 2015-02-03 at 11 26 08 am](https://cloud.githubusercontent.com/assets/1693164/6025360/880027c2-ab97-11e4-8ad9-ff069825ffca.png)

**Comment by bobthecow** [comment]
@una Looks great. :shipit:

**Comment by raghunayyar** [comment]
I can help with developing a CSS style guide btw, have done this before. I know that comes later, just keeping myself in the loop.

**Comment by joeybright** [comment]
Hi all, I'm a designer that's interested in contributing to this project. Before I do, has there been any work on this that hasn't already been posted to this thread? I wouldn't want to re-do any work that's already been completed.

**Comment by una** [comment]
Yay! Go for it!

**Comment by joeybright** [comment]
![mustache_landing_page_v0](https://cloud.githubusercontent.com/assets/4269014/8767826/1ac47690-2e1e-11e5-936b-37e2a819f7de.png)

I jumped right into the design process to riff off of what has already been completed. I decided to post this here (unfinished) to get some initial feedback. If this doesn't work for whatever reason, I can go back to the drawing board and start with some moodboards as @GarthDB suggested. So, if anyone has feedback please post it!

**Comment by una** [comment]
Awesome start!

Things I LOVE:

- the {{}} that open/close the site (so cute, so clever)
- the dark theme code input box
- content heirarchy (title, brief, links to github, example, supported languages)
- I really like the clean feel and typography choices

Additional ideas:

- color-coded a bit better for mustaches syntax in the code box
- link to Github again at the bottom of the page
- allow for the example in multiple languages (a toggle that lets that section show a ruby example, for instance)

**Comment by bobthecow** [comment]
Agreed on all above. The problem with toggling between examples is that this is live code: changing it changes the output. As it is, the example is raw JSON data, not even javascript. It's non-trivial to get ruby (or fake ruby) running in a text box in the browser, especially since this is going to be hosted on github pages, not on a specialized server that can have sandboxes for executing code :)

**Comment by joeybright** [comment]
![mustache_landing_page_v01](https://cloud.githubusercontent.com/assets/4269014/8816106/4ddff8dc-2fd7-11e5-995c-294f5c46cbbb.png)
Progress!
- Added syntax highlighting (let me know if you like the colors/if the syntax highlighting looks correct). Have yet to try toggling between languages here, that's next.
- Added all of the supported languages/implementations. I know that is was mentioned here that it'd be nice to being able to list multiple implementations for each language. However, I don't think that's really feasible, at least for the landing page. My gut feeling is to have each supported language link to a section of the docs that lists all of the implementations in that languages. I think that would keep the landing page lean(er) but allow multiple implementations to be listed.

Feedback is welcome and appreciated!

**Comment by una** [comment]
I don't know how I feel about the grey bar with some of that navigation -- im assuming its sticky?

**Comment by dasilvacontin** [comment]
@bobthecow Why Ruby? Couldn't we use JS for compiling/rendering the templates in the client?

**Comment by bobthecow** [comment]
That was in response to:

> allow for the example in multiple languages (a toggle that lets that section show a ruby example, for instance)

**Comment by dasilvacontin** [comment]
Oh, sorry :smile:. There's not much point on doing that, I think... What for? Showing that some implementations have features not included in the spec? In the spec's website?

**Comment by joeybright** [comment]
@una It would be sticky. Do you not like it's style or do you just not like that it's there?

Here's a version w/o it:
![mustache_landing_page_v01](https://cloud.githubusercontent.com/assets/4269014/8921971/ab362326-3491-11e5-9216-f53190798bd4.png)

**Comment by andresgalante** [comment]
Is this issue still open? If you guys need help to continue with @joeybright work or just to implement it into a html/Css template let me know.

**Comment by raghunayyar** [comment]
Btw, given the current situation of the website and how famous mustache is as a framework, I *really* think that this issue needs to move much faster.

**Comment by bitzl** [comment]
That's true, @raghunayyar. I'd be also available for help (development, testing,...).

**Comment by una** [comment]
I'm a bit over capacity for doing a bulk of work, but I can help with anything you need (design crit, implementation details, etc.) Lets just start making this thing :)

**Comment by andresgalante** [comment]
@raghunayyar @bitzl @una I'll have some bandwidth next week. If it's ok with you guys I'll send a PR with the implementation.

**Comment by locks** [comment]
Hello everyone, thanks for the activity in here!
I thought I had commented a while back, but seems I forgot to actually post. I thought I was going to have a bit of time to start implementing but I didn't get anywhere.
I'm willing to start iterating on this, so we can hopefully address the documentation/implementation-specific page next :)

**Comment by GarthDB** [comment]
I’m doing a workshop at the end of Oct at FOWD on design in the open; We could work on this project during that time if it isn’t already finished.


—

**Comment by joeybright** [comment]
I'm available to make any additions/edits to the landing page if we want to do another few rounds of iteration before implementation. I can also clean up and post the .sketch file for the design when someone starts implementation so they can get their hands on the assets.

**Comment by raghunayyar** [comment]
@joeybright you can upload the sketch file imho. Let's start implementing it. I am very sure implementation will also have a discussion cycle regarding the libraries, grid system, preprocessers  to use and stuff. So, it's better that we just start it right away. I can help out with the implement this weekend itself.

**Comment by joeybright** [comment]
https://www.dropbox.com/s/ffg7yy469r1kiys/mustache.sketch?dl=0 - Dropbox link to the .sketch file.

**Comment by raghunayyar** [comment]
@joeybright awesome, how do I proceed now? Can you add me to the org and I can create a branch or I fork it out, that would be inconvenient though.

**Comment by joeybright** [comment]
I don't have access to the org., so I'd be the wrong person to ask about that. I would imagine if you want to get building sooner than later, forking the repo would make the most sense.

**Comment by raghunayyar** [comment]
@joeybright I am afraid but you will have to upload the zip file with the assets somewhere. My Sketch license is expired. I have clone the repo, will be pushing out changes real soon.

**Comment by locks** [comment]
I have created a `redesign` branch that you can send PRs against so we can get this party started :)
That way I can merge in several PRs, and then merge that into master so the site always works.

**Comment by raghunayyar** [comment]
@locks I have made an initial commit on my fork''s master, just fyi. 
Link : https://github.com/raghunayyar/mustache.github.com
@joeybright I am waiting for the assets. A PSD will also be good.

**Comment by razetime** [comment]
I'm from Open Source Design.
Seems like the website layout is already prepared. 
Why not implement it?

**Comment by dasilvacontin** [comment]
@razetime I guess we are waiting on @raghunayyar, since he already started working on it. He doesn't have Sketch though.

**Comment by joeybright** [comment]
I don't have Photoshop and haven't had time to put together proper specs for @raghunayyar. It's still going to be some time until I'm able to get to that. So, if @raghunayyar is going to be developing the site, he's waiting on me.

If someone wants to either take over development or transfer my .sketch file to a .psd for @raghunayyar, that would speed things up.

**Comment by razetime** [comment]
@joeybright  If you don't have Photoshop, use photoshop CS2, which is free of cost, but old. https://www.adobe.com/support/downloads/detail.jsp?ftpID=3446
Or use the open-source GIMP http://www.gimp.org/, which is similar to Photoshop.

**Comment by raghunayyar** [comment]
@joeybright good news. I got sketch now. I have made a PR but its a WIP. @joeybright can you tell me which highlightJS skin are you using there? Or is it custom?

**Comment by joeybright** [comment]
@raghunayyar Nice! No highlightJS skin used. Just picked colors I liked haha.

**Comment by raghunayyar** [comment]
Btw: here is my pull request again : https://github.com/mustache/mustache.github.com/pull/72. Its ready for review apart from the Try it Out Section. I think I will need someone with better JS skills to get the syntax highlighting in the textareas.

**Comment by viankakrisna** [comment]
hi, i really like the new design for the github page. But can we keep the supported languages at the top? One thing that I really love about mustache is how portable it is. And I think it is a good idea to keep the list of supported language at the top. :)

**Comment by chiraggmodi** [comment]
is this still ongoing? any help needed?

**Comment by una** [comment]
Help is appreciated if you wanna submit some sketches or designs @chiraggmodi :) it looks like this hasn't been worked on it a while

**Comment by chiraggmodi** [comment]
okey will work on some design.

**Comment by pablonoel** [comment]
@chiraggmodi if you need any help, I would love to give you a hand

**Comment by raghunayyar** [comment]
This is a kind of a waste since we decided on a design last year. Built a website for that design, now it lies dead in a lot of PRs which no bothers to review / merge.

**Comment by locks** [comment]
@raghunayyar I'm sorry you feel like your work isn't valued, but things fall on the wayside for a plethora of reasons. I did try finishing your branch at the time by fixing the "Try It Out" but couldn't, so I can't merge it at the moment. Do you know someone who can give you a hand?

**Comment by raghunayyar** [comment]
@locks No, my branch is so old that I have lost context. I am kind of an often open source committer. If would be great if I get commit / review access, specific to repository, I would like to see things accelerate a bit. Your call.

**Comment by Erioldoesdesign** [comment]
Hey there, sorry to dig up an old issue from 2016 but I was wondering if I can close this job on the open source design jobs repo if this has closed or stopped being worked on? https://opensourcedesign.net/jobs/jobs/2015-04-28-mustache

**Comment by jgonggrijp** [comment]
I am able to review and merge pull requests, so if someone wants to have another go at this, please do.

**Comment by Erioldoesdesign** [comment]
> I am able to review and merge pull requests, so if someone wants to have another go at this, please do.

Will make sure the job stays open then and update the forum post :)

**Comment by RayyanNafees** [comment]
After 56 comments over 9 yrs why the doc site still looks crap ?
Please update it with any of those elegant designs above... I can't understand it

**Comment by Erioldoesdesign** [comment]
> I am able to review and merge pull requests, so if someone wants to have another go at this, please do.

I will try to encourage some design help from the opens ource design community - but i think it's stuck more at the FE implementation stage correct?

**Comment by jgonggrijp** [comment]
@Erioldoesdesign please clarify, what is a "FE implementation stage"?

**Comment by jgonggrijp** [comment]
Oh, right. Yes, I think that is the next step and that is also what I meant to invite pull requests for.
